### DISCUSSION TITLE
Using Composite Primary Key Input To findByPk

### DISCUSSION THREAD

**Discussion by liam-ot** [sequelize/sequelize#16869] [Help & Questions]
I did not open this as an issue because I'd like to first know why it is not supported or if maybe it is and im not doing it correctly

I have a table which has a composite primary key (`username` and `id`)

I would like to be able to find them in such a way:
```
await this.sequelizeModel.findByPk({ username, id });
```

but that returns the error:
```
Argument passed to findByPk is invalid: [object Object]
```

Any insight would be appreciated : )

**Comment by ephys** [comment]
It's mostly because when Sequelize was first designed, it was designed without composite PK in mind. A few places still need to be updated to support them.

One of the issues with `findByPk` is that we can't type it correctly. It's input will always be `unknown`, so TypeScript won't be able to catch issues like typos in property names. That's a general issue with `findByPk` though, so I don't think it should prevent adding support for composite PKs there
